src/analysis/override_string_type.rs

```diff
@@ -34,11 +34,11 @@ fn apply(env: &Env, type_id: TypeId, string_type: Option<config::StringType>) ->
             OsString => TypeId::tid_os_string(),
         }
     };
-    match *env.library.type_(type_id) {
+    match env.library.type_(type_id) {
         Type::Fundamental(Fundamental::Filename) => replace,
         Type::Fundamental(Fundamental::OsString) => replace,
         Type::Fundamental(Fundamental::Utf8) => replace,
-        Type::CArray(inner_tid) if can_overriden_fundamental(env, inner_tid) => {
+        &Type::CArray(inner_tid) if can_overriden_fundamental(env, inner_tid) => {
```

Comment:
Also here

Reply:
Same here, you cannot "destructure" the borrow with `Type::CArray(&inner_tid)` - note that this is _the inverse of_ `ref` :)

Reply:
Alternative is to write `*inner_tid` and friends everywhere.